xdsl/dialects/csl.py

```diff
@@ -703,6 +740,16 @@ class SetTileCodeOp(IRDLOperation):
     y_coord = operand_def(IntegerType)
     params = opt_operand_def(ComptimeStructType)
 
+    def __init__(
+        self,
+        fname: str | StringAttr,
+        x_coord: SSAValue | Operation,
+        y_coord: SSAValue | Operation,
+        params: SSAValue | Operation,
```

Comment:
This should be `SSAValue | Operation | None` as it's optional

Reply:
done as suggested